--- boost/build-android.sh	2019-04-08 21:20:28.237478475 +0200
+++ boost/build-android.sh	2019-04-08 21:22:01.801445330 +0200
@@ -208,6 +208,11 @@
 
 echo "Detected Android NDK version $NDK_RN"
 
+cxxprefix=arm-linux-androideabi
+if [ "$TOOLCHAIN" = "x86_64-4.9" ] ; then
+  cxxprefix=x86_64-linux-android
+fi
+
 case "$NDK_RN" in
 	4*)
 		TOOLCHAIN=${TOOLCHAIN:-arm-eabi-4.4.0}
@@ -215,38 +220,43 @@
 		TOOLSET=gcc-androidR4
 		;;
 	5*)
-		TOOLCHAIN=${TOOLCHAIN:-arm-linux-androideabi-4.4.3}
-		CXXPATH=$AndroidNDKRoot/toolchains/${TOOLCHAIN}/prebuilt/$PlatformOS-x86/bin/arm-linux-androideabi-g++
+		TOOLCHAIN=${TOOLCHAIN:-${cxxprefix}-4.4.3}
+		CXXPATH=$AndroidNDKRoot/toolchains/${TOOLCHAIN}/prebuilt/$PlatformOS-x86/bin/${cxxprefix}-g++
 		TOOLSET=gcc-androidR5
 		;;
 	7-crystax-5.beta3)
-		TOOLCHAIN=${TOOLCHAIN:-arm-linux-androideabi-4.6.3}
-		CXXPATH=$AndroidNDKRoot/toolchains/${TOOLCHAIN}/prebuilt/$PlatformOS-x86/bin/arm-linux-androideabi-g++
+		TOOLCHAIN=${TOOLCHAIN:-${cxxprefix}-4.6.3}
+		CXXPATH=$AndroidNDKRoot/toolchains/${TOOLCHAIN}/prebuilt/$PlatformOS-x86/bin/${cxxprefix}-g++
 		TOOLSET=gcc-androidR7crystax5beta3
 		;;
 	8)
-		TOOLCHAIN=${TOOLCHAIN:-arm-linux-androideabi-4.4.3}
-		CXXPATH=$AndroidNDKRoot/toolchains/${TOOLCHAIN}/prebuilt/$PlatformOS-x86/bin/arm-linux-androideabi-g++
+		TOOLCHAIN=${TOOLCHAIN:-${cxxprefix}-4.4.3}
+		CXXPATH=$AndroidNDKRoot/toolchains/${TOOLCHAIN}/prebuilt/$PlatformOS-x86/bin/${cxxprefix}-g++
 		TOOLSET=gcc-androidR8
 		;;
 	8b|8c|8d)
-		TOOLCHAIN=${TOOLCHAIN:-arm-linux-androideabi-4.6}
-		CXXPATH=$AndroidNDKRoot/toolchains/${TOOLCHAIN}/prebuilt/$PlatformOS-x86/bin/arm-linux-androideabi-g++
+		TOOLCHAIN=${TOOLCHAIN:-${cxxprefix}-4.6}
+		CXXPATH=$AndroidNDKRoot/toolchains/${TOOLCHAIN}/prebuilt/$PlatformOS-x86/bin/${cxxprefix}-g++
 		TOOLSET=gcc-androidR8b
 		;;
 	8e|9|9b|9c|9d)
-		TOOLCHAIN=${TOOLCHAIN:-arm-linux-androideabi-4.6}
-		CXXPATH=$AndroidNDKRoot/toolchains/${TOOLCHAIN}/prebuilt/$PlatformOS-x86/bin/arm-linux-androideabi-g++
+		TOOLCHAIN=${TOOLCHAIN:-${cxxprefix}-4.6}
+		CXXPATH=$AndroidNDKRoot/toolchains/${TOOLCHAIN}/prebuilt/$PlatformOS-x86/bin/${cxxprefix}-g++
 		TOOLSET=gcc-androidR8e
 		;;
 	"8e (64-bit)")
-		TOOLCHAIN=${TOOLCHAIN:-arm-linux-androideabi-4.6}
-		CXXPATH=$AndroidNDKRoot/toolchains/${TOOLCHAIN}/prebuilt/${PlatformOS}-x86_64/bin/arm-linux-androideabi-g++
+		TOOLCHAIN=${TOOLCHAIN:-${cxxprefix}-4.6}
+		CXXPATH=$AndroidNDKRoot/toolchains/${TOOLCHAIN}/prebuilt/${PlatformOS}-x86_64/bin/${cxxprefix}-g++
 		TOOLSET=gcc-androidR8e
 		;;
 	"9 (64-bit)"|"9b (64-bit)"|"9c (64-bit)"|"9d (64-bit)")
-		TOOLCHAIN=${TOOLCHAIN:-arm-linux-androideabi-4.6}
-		CXXPATH=$AndroidNDKRoot/toolchains/${TOOLCHAIN}/prebuilt/${PlatformOS}-x86_64/bin/arm-linux-androideabi-g++
+		TOOLCHAIN=${TOOLCHAIN:-${cxxprefix}-4.6}
+		CXXPATH=$AndroidNDKRoot/toolchains/${TOOLCHAIN}/prebuilt/${PlatformOS}-x86_64/bin/${cxxprefix}-g++
+		TOOLSET=gcc-androidR8e
+		;;
+	"10e-rc4 (64-bit)")
+		TOOLCHAIN=${TOOLCHAIN:-${cxxprefix}-4.9}
+		CXXPATH=$AndroidNDKRoot/toolchains/${TOOLCHAIN}/prebuilt/${PlatformOS}-x86_64/bin/${cxxprefix}-g++
 		TOOLSET=gcc-androidR8e
 		;;
 	*)
@@ -255,7 +265,11 @@
 esac
 
 if [ -n "${AndroidSourcesDetected}" ]; then # Overwrite CXXPATH if we are building from Android sources
-    CXXPATH="${ANDROID_TOOLCHAIN}/arm-linux-androideabi-g++"
+    CXXPATH="${ANDROID_TOOLCHAIN}/${cxxprefix}-g++"
+fi
+
+if [ "$TOOLCHAIN" = "x86_64-4.9" ] ; then
+  TOOLSET=${TOOLSET}x86_64
 fi
 
 echo Building with TOOLSET=$TOOLSET CXXPATH=$CXXPATH CXXFLAGS=$CXXFLAGS | tee $PROGDIR/build.log
@@ -370,6 +384,7 @@
   cxxflags=""
   for flag in $CXXFLAGS; do cxxflags="$cxxflags cxxflags=$flag"; done
 
+
   { ./bjam -q                         \
          toolset=$TOOLSET             \
          $cxxflags                    \
