--- libs/libxml2-2.9.1/Makefile.in	2018-09-23 19:41:02.131676136 +0200
+++ libs/libxml2-2.9.1/Makefile.in	2018-09-23 19:41:31.391661888 +0200
@@ -125,7 +125,7 @@
 	parserInternals.c parser.c tree.c hash.c list.c xmlIO.c \
 	xmlmemory.c uri.c valid.c xlink.c HTMLparser.c HTMLtree.c \
 	debugXML.c xpath.c xpointer.c xinclude.c nanohttp.c nanoftp.c \
-	DOCBparser.c catalog.c globals.c threads.c c14n.c xmlstring.c \
+	DOCBparser.c catalog.c glob.c globals.c threads.c c14n.c xmlstring.c \
 	buf.c xmlregexp.c xmlschemas.c xmlschemastypes.c xmlunicode.c \
 	triostr.c trio.c xmlreader.c relaxng.c dict.c SAX2.c \
 	xmlwriter.c legacy.c chvalid.c pattern.c xmlsave.c xmlmodule.c \
@@ -136,7 +136,7 @@
 	parserInternals.lo parser.lo tree.lo hash.lo list.lo xmlIO.lo \
 	xmlmemory.lo uri.lo valid.lo xlink.lo HTMLparser.lo \
 	HTMLtree.lo debugXML.lo xpath.lo xpointer.lo xinclude.lo \
-	nanohttp.lo nanoftp.lo $(am__objects_1) catalog.lo globals.lo \
+	nanohttp.lo nanoftp.lo $(am__objects_1) catalog.lo glob.lo globals.lo \
 	threads.lo c14n.lo xmlstring.lo buf.lo xmlregexp.lo \
 	xmlschemas.lo xmlschemastypes.lo xmlunicode.lo \
 	$(am__objects_2) xmlreader.lo relaxng.lo dict.lo SAX2.lo \
@@ -634,7 +634,7 @@
 		valid.c xlink.c HTMLparser.c HTMLtree.c debugXML.c xpath.c  \
 		xpointer.c xinclude.c nanohttp.c nanoftp.c \
 		$(docb_sources) \
-		catalog.c globals.c threads.c c14n.c xmlstring.c buf.c \
+		catalog.c glob.c globals.c threads.c c14n.c xmlstring.c buf.c \
 		xmlregexp.c xmlschemas.c xmlschemastypes.c xmlunicode.c \
 		$(trio_sources) \
 		xmlreader.c relaxng.c dict.c SAX2.c \
@@ -1058,6 +1058,7 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/entities.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/error.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/globals.Plo@am__quote@
+@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/glob.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/hash.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/legacy.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/list.Plo@am__quote@
